backstory/src/test/java/com/google/sps/images/AnnotatedImageTest.java

```diff
@@ -0,0 +1,177 @@
+import java.util.ArrayList;
+import java.util.List;
+import java.lang.IllegalArgumentException;
+import java.io.IOException;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import org.junit.Test;
+import java.io.ByteArrayOutputStream;
+import java.awt.image.BufferedImage;
+import java.io.File;
+import java.util.Arrays;
+import javax.imageio.ImageIO;
+import java.lang.IllegalArgumentException;
+import com.google.sps.images.data.AnnotatedImage;
+
+/**
+ * Tests for AnnotatedImage.
+ * Testing strategy for each operation of AnnotatedImage:
+ *
+ * public AnnotatedImage(byte[] rawImageData, List<EntityAnnotation> labelAnnotations);
+ *  Partition on rawImageData: null, empty, non-empty image data.
+ *  Partition on labelAnnotations: empty, non-empty image data.
+ * 
+ * public String getLabelsAsJson();
+ *  Unnecesary to test as it is a wrapper for a Gson call.
+ *
+ * public List<String> getLabelDescriptions();
+ *  Partition on this: labelAnnotations is empty, or non-empty list of EntityAnnotations.
+ */
+public final class AnnotatedImageTest {
+  private static final byte[] nullRawImageData = null;
+  private static final byte[] emptyRawImageData = new byte[0];
+  private static final List<EntityAnnotation> emptyLabelAnnotations = new ArrayList<>();
+
+  /**
+   * Create a mock entity annotation with a description field set.
+   *
+   * @param description the description to give the mock entity annotation
+   * @return the mock entity annotation
+   */
+  private static EntityAnnotation makeMockEntityAnnotationWithDescription(String description) {
+    EntityAnnotation mockEntityAnnotation = mock(EntityAnnotation.class);
+    when(mockEntityAnnotation.getDescription()).thenReturn(description);
+    return mockEntityAnnotation;
+  }
+
+  /**
+   * Returns an image as a byte array from the local reference of the image.
+   *
+   * @param reference the local reference of the image.
+   * @param fileType the file extension of the image.
+   * @return the byte array representation of the image.
+   */
+  private static byte[] getBytesFromImageReference(String reference, String fileType) throws IOException {
+    ByteArrayOutputStream bos = new ByteArrayOutputStream();
+
+    try {
+      BufferedImage bImage = ImageIO.read(new File(reference));
+      ImageIO.write(bImage, fileType, bos);
+    } catch (IOException exception){
+      throw exception;
+    }
+
+    return bos.toByteArray();
+  }
+
+  /**
+   * public AnnotatedImage(byte[] rawImageData, List<EntityAnnotation> labelAnnotations);
+   *  rawImageData is null.
+   *  labelAnnotations is empty.
+   *  Expects IllegalArgumentException to be thrown.
+  */
+  @Test(expected = IllegalArgumentException.class)
+  public void constructorNullImageData() throws IllegalArgumentException {
+    AnnotatedImage annotatedImageExpectingException = new AnnotatedImage(nullRawImageData, emptyLabelAnnotations);
```

Comment:
As mentioned in another comment, may be clearer to just not assign to a local variable, if the variable is unused:

```
public void constructor... {
  new AnnotatedImage(...);
}
```

Reply:
Updated.